src/PhpParser/NodeVisitor/ParentClassNameCollectingNodeVisitor.php

```diff
@@ -39,14 +39,8 @@ public function getParentClassNames(): array
         $uniqueParentClassNames = array_unique($this->parentClassNames);
         sort($uniqueParentClassNames);
 
-        // remove native classes
-        $namespacedClassNames = array_filter(
-            $uniqueParentClassNames,
-            static fn (string $parentClassName): bool => str_contains($parentClassName, '\\')
```

Comment:
to sum up: the bug this PR is fixing is this code here filters out way more classes then just "native classes"